rspace/src/test/scala/coop/rchain/rspace/history/RadixTreeSpec.scala

```diff
@@ -0,0 +1,726 @@
+
+        newRootNode    <- impl.constructNodeFromItem(item1.get)
+        printedTreeStr <- impl.printTree(newRootNode, "TREE WITH ONE LEAF", true)
+
+        etalonTree = Vector(
+          "TREE WITH ONE LEAF: root =>",
+          "   [FF]LEAF: prefix = FFFFF1, data = 0000...000A"
+        )
+
+        _ = printedTreeStr shouldBe etalonTree
+      } yield ()
+  }
+
+  "appending leaf to tree with one leaf " should "create 2 leafs with node ptr" in withImplAndStore {
+    (impl, _) =>
+      val commonKeyPartForTwoLeafs = "001122"
+      val keys = Vector[Seq[Byte]](
+        TestData.hexKey(commonKeyPartForTwoLeafs + "013"),
+        TestData.hexKey(commonKeyPartForTwoLeafs + "225")
+      )
+      for {
+        item1Opt <- impl.update(
+                     RadixTree.EmptyItem,
+                     keys(0),
+                     createBV32(0x11.toByte)
+                   )
+
+        rootNode1    <- impl.constructNodeFromItem(item1Opt.get)
+        printedTree1 <- impl.printTree(rootNode1, "TREE WITH ONE LEAF", true)
+
+        etalonTree1 = Vector(
+          "TREE WITH ONE LEAF: root =>",
+          "   [00]LEAF: prefix = 01122013, data = 0000...0011"
+        )
+
+        item2Opt <- impl.update(
+                     item1Opt.get,
+                     keys(1),
+                     createBV32(0x55.toByte)
+                   )
+
+        rootNode2    <- impl.constructNodeFromItem(item2Opt.get)
+        printedTree2 <- impl.printTree(rootNode2, "TREE WITH ONE NODE AND 2 LEAFS", true)
+
+        etalonTree2 = Vector(
+          "TREE WITH ONE NODE AND 2 LEAFS: root =>",
+          "   [00]PTR: prefix = 0112, ptr =>",
+          "      [20]LEAF: prefix = 13, data = 0000...0011",
+          "      [22]LEAF: prefix = 25, data = 0000...0055"
+        )
+
+        _ = printedTree1 shouldBe etalonTree1
+        _ = printedTree2 shouldBe etalonTree2
+      } yield ()
+  }
+  "appending leaf to leaf" should "create node with two leafs" in withImplAndStore { (impl, _) =>
+    val keys = Vector[Seq[Byte]](
+      TestData.hexKey("00000000"),
+      TestData.hexKey("34564544")
+    )
+    for {
+      rootItem1Opt <- impl.update(
+                       RadixTree.EmptyItem,
+                       keys(0),
+                       createBV32(0x11.toByte)
+                     )
+
+      rootItem2Opt <- impl.update(
+                       rootItem1Opt.get,
+                       keys(1),
+                       createBV32(0x55.toByte)
+                     )
+
+      rootNode <- impl.constructNodeFromItem(rootItem2Opt.get)
+
+      printedTreeStr <- impl.printTree(rootNode, "TREE: TWO LEAFS", true)
+
+      etalonTree = Vector(
+        "TREE: TWO LEAFS: root =>",
+        "   [00]LEAF: prefix = 000000, data = 0000...0011",
+        "   [34]LEAF: prefix = 564544, data = 0000...0055"
+      )
+
+      _ = printedTreeStr shouldBe etalonTree
+    } yield ()
+  }
+
+  "updating leaf" should "work correctly" in withImplAndStore { (impl, _) =>
```

Comment:
If possible. test description should say exactly what is doing. "Works correctly" should be the last resort.